src/main/java/com/microsoft/jenkins/containeragents/builders/AciDeploymentTemplateBuilder.java

```diff
@@ -0,0 +1,341 @@
+import com.microsoft.jenkins.containeragents.aci.AciCloud;
+import com.microsoft.jenkins.containeragents.aci.AciContainerTemplate;
+import com.microsoft.jenkins.containeragents.aci.AciPort;
+import com.microsoft.jenkins.containeragents.aci.AciPrivateIpAddress;
+import com.microsoft.jenkins.containeragents.aci.AciService;
+import com.microsoft.jenkins.containeragents.aci.volumes.AzureFileVolume;
+import com.microsoft.jenkins.containeragents.util.AzureContainerUtils;
+import com.microsoft.jenkins.containeragents.util.Constants;
+import com.microsoft.jenkins.containeragents.util.DockerRegistryUtils;
+import hudson.EnvVars;
+import hudson.security.ACL;
+import hudson.slaves.SlaveComputer;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.jenkinsci.main.modules.instance_identity.InstanceIdentity;
+import org.jenkinsci.plugins.docker.commons.credentials.DockerRegistryEndpoint;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class AciDeploymentTemplateBuilder {
+
+    private static final String DEPLOY_TEMPLATE_FILENAME
+            = "/com/microsoft/jenkins/containeragents/aci/deployTemplate.json";
+    private static final String NETWORK_PROFILE_SNIPPET_FILENAME
+            = "/com/microsoft/jenkins/containeragents/aci/networkProfileSnippet.json";
+    private AciDeploymentTemplateBuilder() {
+        //
+    }
+
+    @NotNull
+    public static AciDeploymentTemplate buildDeploymentTemplate(AciCloud cloud, AciContainerTemplate template,
+                                                                AciAgent agent) throws IOException {
+        try (InputStream stream = AciService.class.getResourceAsStream(DEPLOY_TEMPLATE_FILENAME)) {
+            AciPrivateIpAddress privateIpAddress = template.getPrivateIpAddress();
+
+            final ObjectMapper mapper = new ObjectMapper();
+            final JsonNode tmp = mapper.readTree(stream);
+            final ObjectNode parameters = mapper.createObjectNode();
+
+            ObjectNode variables = (ObjectNode) tmp.get("variables");
+            variables.put("containerName", agent.getNodeName());
+            variables.put("containerImage", template.getImage());
+            variables.put("osType", template.getOsType());
+            variables.put("ipType", mapIpType(privateIpAddress));
+            if (privateIpAddress != null) {
+                variables.put("vnetName", privateIpAddress.getVnet());
+                variables.put("subnetName", privateIpAddress.getSubnet());
+            }
+            variables.put("cpu", template.getCpu());
+            variables.put("memory", template.getMemory());
+            variables.put("jenkinsInstance",
+                    Jenkins.get().getLegacyInstanceId());
+
+            addLogAnalytics(tmp, parameters, mapper, cloud);
```

Comment:
`addPortNode` and `addLogAnalytics` differ in guards, one is guarded in this method and one is guarded inside the `addLogAnalytics` is there a reason for this?

I think it would be slightly clearer that it is optional if the `if` check was pulled up here

Reply:
Sorry, I don't know. Because this code is older than my changes and I refactored it to a new class.

Reply:
That’s fine not a blocker or anything